### PR TITLE
Rename open telemetry traces

### PR DESCRIPTION
**Pull request by Stefan-Ethernal**
There were some duplicate trace names (`ValidateState`) which was confusing. Now all the traces are named consistent and uniquely, so their name suggests its purpose.

Although maybe we should consider not having separate traces for message reading loops in `AcceptState`, `ValidateState` and `RoundChangeState`, but those traces to be integral part of given state trace?

### CODE REVIEW

File: consensus_pbft.go [hunk 1]
```diff
@@ -741,9 +751,9 @@ func (p *Pbft) getNextMessage(span trace.Span) (*MessageReq, bool) {
 			p.spanAddEventMessage("dropMessage", span, msg)
 		}
 		if msg != nil {
+			p.logger.Printf("[TRACE] Received %s", msg)
```
Comment (ferranbt) [inline consensus_pbft.go:754]: Why did you move this?

File: consensus_pbft.go [hunk 2]
```diff
@@ -387,19 +393,21 @@ func (p *Pbft) runValidateState(ctx context.Context) { // start new round
 	}
 
 	quorum := p.state.getQuorumSize()
+	i := 1
 	for p.getState() == ValidateState {
-		_, span := p.tracer.Start(ctx, "ValidateState")
+		_, loopSpan := p.tracer.Start(ctx, fmt.Sprintf("ValidateState_Read#%d", i))
```
Comment (ferranbt) [inline consensus_pbft.go:398]: To me, this counter should be metadata of the trace, not part of the name itself.

File: consensus_pbft.go [hunk 3]
```diff
@@ -321,13 +321,19 @@ func (p *Pbft) runAcceptState(ctx context.Context) { // start new round
 	// We only need to wait here for one type of message, the Prepare message from the proposer.
 	// However, since we can receive bad Prepare messages we have to wait (or timeout) until
 	// we get the message from the correct proposer.
+	i := 1
```
Comment (ferranbt) [inline consensus_pbft.go:324]: Please rename this to `counter`.

File: consensus_pbft.go [hunk 4]
```diff
@@ -581,9 +589,11 @@ func (p *Pbft) runRoundChangeState(ctx context.Context) {
 		}
 	}
 
+	i := 1
 	// create a timer for the round change
 	for p.getState() == RoundChangeState {
-		_, span := p.tracer.Start(ctx, "RoundChangeState")
+		_, span := p.tracer.Start(ctx, fmt.Sprintf("RoundChange_Read#%d", i))
```
Comment (ferranbt) [inline consensus_pbft.go:595]: This variable gets called `span` and `loopSpan`. I would prefer to call it `span` always since there are no other spans around.

**Review by ferranbt — CHANGES_REQUESTED** [review]

File: consensus_pbft.go (hunk 1 above)
Comment (Stefan-Ethernal) [reply consensus_pbft.go:754]: Just wanted to make it consistent with the for loop above: https://github.com/0xPolygon/pbft-consensus/pull/76/files#diff-b208c241c10c010125c6a1c4f43ac407c5fb7b363bab10a050e5681019875478R750-R751. There we are first logging and than reporting metadata to the tracer, so wanted to keep it the same in this block as well.

I can revert it back,  if it bothers you.

File: consensus_pbft.go (hunk 2 above)
Comment (Stefan-Ethernal) [reply consensus_pbft.go:398]: Is it ok to add attribute to the span this way or you have different idea?
`span.SetAttributes(attribute.Int("loop.counter", counter))`

I mean maybe we don't even need this `counter` at all. Intention was just to differ span within for loop with this outer span for handling specific state.

File: consensus_pbft.go [hunk 5]
```diff
@@ -435,11 +433,6 @@ func (p *Pbft) runValidateState(ctx context.Context) { // start new round
 			// change to commit state just to get out of the loop
 			p.setState(CommitState)
 		}
-
-		// set the attributes of this span once it is done
-		p.setStateSpanAttributes(span)
```
Comment (ferranbt) [inline consensus_pbft.go:440]: This might be the reason why there was an inter-span. Imagine that the system gets locked in one of the cycles, we would not get any traces at all because it is waiting to exit.

**Review by ferranbt — CHANGES_REQUESTED** [review]

File: consensus_pbft.go (hunk 5 above)
Comment (Stefan-Ethernal) [reply consensus_pbft.go:440]: Are you referring that `runXXXState` (`runValidateState`, `runRoundChangeState` etc.) execution could stuck? How is that even possible (at least with the current implementation)? Eventually a timeout would elapse and the execution of a  PBFT state handler would come to an end.
If PBFT consensus is stuck (or locked as you are referring) it still means that we would have transitions as such: `AcceptState` -> `RoundChangeState` -> `AcceptState` etc.

So do you still find useful leaving inter-spans or not?

Comment (vcastellm) [reply consensus_pbft.go:440]: Can you elaborate on this?

Comment (Stefan-Ethernal) [reply consensus_pbft.go:440]: > Can you elaborate on this?

My point was to keep only single span through entire PBFT state handler (instead of reporting spans every time a message is read from the queue). 
Both approaches have positive and negative aspects: 
- **creating a span for each message retrieval** (like it was before) 
   pros: we are sure that we wouldn't miss any information, if (for some strange reason), we get stuck any of the `runXXXState` function.
   cons: in order to track which messages are aggregated through given PBFT state, we would need to analyze 
   separately each span, which can be tedious, since we don't have reported all the accumulated messages at the single 
   place, but reporting it one by one.
- **creating a single span for entire PBFT state handler**
  pros: all the information are aggregated and easily consumed
  cons: Ferran's concern [here](https://github.com/0xPolygon/pbft-consensus/pull/76#discussion_r940043739). Although I   don't see that it is likely to happen (as we have timeouts as a mechanism to break execution if it lasts too long).
  
On the end, I'm good with both approaches, just need to make an agreement and I'll accommodate it.

Comment (vcastellm) [reply consensus_pbft.go:440]: Yes I like this approach @Stefan-Ethernal, I was asking @ferranbt to explain his concerns 😄 because as you said, I don't see any stopper

Comment (ferranbt) [reply consensus_pbft.go:440]: As discussed, lets change this except for RoundChange state.

**Review by begmaroman — APPROVED** [review]

**Review by vcastellm — APPROVED** [review]

**Review by ferranbt — APPROVED** [review]
